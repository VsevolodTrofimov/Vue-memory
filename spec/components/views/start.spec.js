import Vuex from 'vuex'
import { mount, createLocalVue } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'

import StartVeiw from '@/src/components/views/Start/Start.vue'


const VSSR = createRenderer()
const localVue = createLocalVue()
localVue.config.productionTip = false
localVue.use(Vuex)

describe('Views', () => {
  describe('Start View', () => {
    let actions
    let stubStore
    let stage

    beforeEach(() => {
      stage = 'start'

      actions = {
        startGame: jest.fn(),
        loadDeck: jest.fn()
      }
      
      stubStore = new Vuex.Store({
        state: {},
        getters: {
          stage: () => {
            return stage
          }
        },
        actions
      })
    })

    it('matches snapshot', () => {
      const wrapper = mount(StartVeiw, {store: stubStore, localVue})

      VSSR.renderToString(wrapper.vm, (err, str) => {
        expect(str).toMatchSnapshot()
      })
    })
    
    it('dispatches startGame action', () => {
      const wrapper = mount(StartVeiw, {store: stubStore, localVue})
      wrapper.find('[data-tid="NewGame-startGame"]').trigger('click')

      expect(actions.startGame).toBeCalled()
    })

    it('dispatches loadDeck action', () => {
      const wrapper = mount(StartVeiw, {store: stubStore, localVue})
      wrapper.find('[data-tid="NewGame-startGame"]').trigger('click')

      expect(actions.loadDeck).toBeCalled()
    })

    it('matches "loading" snapshot', () => {
      stage = 'loading'

      const vm = new localVue({
        store: stubStore,
        render: h => h(StartVeiw)
      })

      VSSR.renderToString(vm, (err, str) => {
        expect(str).toMatchSnapshot()
      })
    })
  })
})