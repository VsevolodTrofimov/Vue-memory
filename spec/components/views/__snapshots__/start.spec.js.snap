// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start View matches "loading" snapshot 1`] = `
<div class="view">
    <img srcset="/static/banners/StartGame.png, /static/banners/StartGame@2x.png 2x" alt="Flipped cards" class="banner">
    <h1 class="heading title"> Memory Game </h1>
    <button disabled="disabled" class="btn">
        Раскладываем карты...
    </button>
</div>
`;

exports[`Start View matches snapshot 1`] = `
<div class="view">
    <img srcset="/static/banners/StartGame.png, /static/banners/StartGame@2x.png 2x" alt="Flipped cards" class="banner">
    <h1 class="heading title"> Memory Game </h1>
    <button data-tid="NewGame-startGame" class="btn"> Начать игру </button>
</div>
`;
